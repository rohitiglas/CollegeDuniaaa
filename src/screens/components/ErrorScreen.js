import React from 'react';
import {View, Text, Button} from 'react-native';

const Error = ({navigation, retryButton}) => {
  return (
    <View
      style={{height: '90%', alignItems: 'center', justifyContent: 'center'}}>
      <Text style={{fontSize: 24, textAlign: 'center'}}>
        Something went wrong at our end
      </Text>
      <Button onPress={retryButton} title="Retry" />
    </View>
  );
};

export default Error;
