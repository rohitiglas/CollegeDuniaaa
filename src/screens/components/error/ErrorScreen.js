import React from 'react';
import {View, Text, Button} from 'react-native';

const Error = ({retryButton}) => {
  return (
    <View
      style={{height: '90%', alignItems: 'center', justifyContent: 'center'}}>
      <Text testID="error-text" style={{fontSize: 24, textAlign: 'center'}}>
        Something went wrong at our end
      </Text>
      <Button testID="button-retry" onPress={retryButton} title="Retry" />
    </View>
  );
};

export default Error;
