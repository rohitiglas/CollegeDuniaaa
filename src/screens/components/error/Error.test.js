import 'react-native';
import React from 'react';
import ErrorScreen from './ErrorScreen';
import {render, cleanup, fireEvent} from '@testing-library/react-native';
describe('Error Screen', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = render(<ErrorScreen />);
  });
  afterEach(() => {
    cleanup();
    wrapper = null;
  });
  it('should render ErrorScreen correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show Error text and button', () => {
    const {getByTestId} = wrapper;
    expect(getByTestId('error-text').props.children).toStrictEqual(
      'Something went wrong at our end',
    );
    const submitButton = getByTestId('button-retry');
    fireEvent(submitButton, 'onClick');
  });
});
