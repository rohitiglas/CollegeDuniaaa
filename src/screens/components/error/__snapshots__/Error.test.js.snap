// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Screen should render ErrorScreen correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "90%",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
    testID="error-text"
  >
    Something went wrong at our end
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button-retry"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Retry
      </Text>
    </View>
  </View>
</View>
`;

exports[`Login Form renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": "90%",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
    testID="error-text"
  >
    Something went wrong at our end
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button-retry"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Retry
      </Text>
    </View>
  </View>
</View>
`;
